kotsu/run.py

```diff
@@ -27,13 +29,18 @@ def run(
             each model through.
         store_directory: A file path or URI location to store the validation results and any extra
             output artefacts of the validations and models.
+        pass_artefacts_directory: Flag, if True then Validations will be passed an
+            `artefacts_directory: str` kwarg along with Model arg.
         run_params: A dictionary of optional run parameters.
     """
     results_list = []
     for validation_spec in validation_registry.all():
         for model_spec in model_registry.all():
             logger.info(f"Running validation - model: {validation_spec.id} - {model_spec.id}")
             validation: Validation = validation_spec.make()
+            if pass_artefacts_directory is True:
+                artefacts_directory = store_directory + f"{validation_spec.id}/{model_spec.id}/"
+                validation = functools.partial(validation, artefacts_directory=artefacts_directory)
```

Comment:
potentially we could _always_ pass a `context` dictionary (a bit like airflow's Context that they attach to all the TaskInstances) and demand that validations accept it? which can be a place to store any context about the validation run (ie, in this case, where artefacts should be stored, but we can easily add to it in future and keep it documented easily without having to keep changing the API)

also saves some work with allowing different `validation` types and needing more `run` arguments

Reply:
Yea I I think this is good. Cool if we add this once we (inevitably) need to add another piece of context beyond just `artefacts_directory`?

Reply:
Added it to #19 also